File: Data Structure and Algorithm/Assignment 01/Program 07/prog7.c
//Write a C program to merge two sorted dynamic array

#include<stdio.h>
#include<stdlib.h>

void initialize (int *arr,int n){
    for (int i = 0; i < n; i++)
    {
        printf("Enter the element %d: ",i+1);
        scanf("%d",&arr[i]);
    }
}

int main(){
    int n ,m;
    printf("Enter the number of inputs for first array: ");
    scanf("%d",&n);
    int *arr1 = (int*)malloc(n*sizeof(int));
    initialize (arr1,n);


    printf("Enter the number of inputs for second array: ");
    scanf("%d",&m);
    int *arr2 = (int*)malloc(m*sizeof(int));
    initialize (arr2,m);
    
    int *sorted_arr = (int*)malloc((n+m)*sizeof(int));
    int k = 0,l = 0,d=0;
    while (d < m && l < n)
    {
        if(arr1[d] < arr2[l]){
            sorted_arr[k++] = arr1[d++];
        }
        else{
            sorted_arr[k++] = arr2[l++];
        }
    }
    
    while (d < m)
    {
        sorted_arr[k++] = arr1[d++];
    }

    while (l < n)
    {
        sorted_arr[k++] = arr2[l++];
    }
    
    

    for (int i = 0; i < k; i++)
    {
        printf("%d ",sorted_arr[i]);
    }
    
    free(arr1);
    free(arr2);
    free(sorted_arr);
    return 0;
}

File: Assignment 01/Program 07/prog7.py
#Write a C program to merge two sorted dynamic array

import array as arr

n = int(input("Enter the no. of inputs of first array: "))
a1 = arr.array('i')
for i in range(n):
    val = int(input(f"Enter the element {i+1}:"))
    a1.append(val)

m = int(input("Enter the no. of inputs of second array: "))
a2 = arr.array('i')
for j in range(n):
    val = int(input(f"Enter the element {j+1}:"))
    a2.append(val)

a3 = arr.array('i')
l,d = 0,0

while l<len(a1) and d<len(a2):
    if a1[l] < a2[d]:
        a3.append(a1[l])
        l+=1
    else:
        a3.append(a2[d])
        d+=1

while l<len(a1):
    a3.append(a1[l])
    l+=1

while d<len(a2):
    a3.append(a2[d])
    d+=1

print(a3)